File: Bridge/cpp-source/axles/PlaneAxle.cpp
//
// Created by Pavel Akhtyamov on 2019-03-26.
//

#include <PlaneAxle.h>
#include <iostream>


void PlaneAxle::Accelerate(float speed) {
  speed_ += speed;
}

void PlaneAxle::Brake(float speed) {
  speed_ -= speed;
  if (speed_ < 0) {
    speed_ = 0;
  }
}

void PlaneAxle::MoveLeft(float angle) {
  rotate_angle_ += angle;
}

void PlaneAxle::MoveRight(float angle) {
  rotate_angle_ -= angle;
}

void PlaneAxle::MoveUp(float height) {
  angle_ += height;
}

void PlaneAxle::MoveDown(float height) {
  angle_ -= height;
  if (angle_ < 0) {
    angle_ = 0;
  }
}

void PlaneAxle::Beep() {

}

void PlaneAxle::PrintInfo() {
  std::cout << "Type: Plane" << std::endl;
  std::cout << "Speed: " << speed_ << std::endl;
  std::cout << "Angle: " << angle_ << std::endl;
  std::cout << "Rotate angle: " << rotate_angle_ << std::endl;
}

File: Bridge/python-source/axles/PlaneAxle.py
from __future__ import annotations

from .Axle import Axle


class PlaneAxle(Axle):
    """Самолёт умеет всё то же самое плюс высоту."""

    def __init__(self) -> None:
        self._speed = 0.0
        self._rotate_angle = 0.0
        self._height = 0.0

    def accelerate(self, speed: float) -> None:
        self._speed += speed

    def brake(self, speed: float) -> None:
        self._speed = max(0.0, self._speed - speed)

    def move_left(self, angle: float) -> None:
        self._rotate_angle += angle

    def move_right(self, angle: float) -> None:
        self._rotate_angle -= angle

    def move_up(self, height: float) -> None:
        self._height += height

    def move_down(self, height: float) -> None:
        self._height = max(0.0, self._height - height)

    def beep(self) -> None:
        print("Whoosh")

    def print_info(self) -> None:
        print("Type: Plane")
        print(f"Speed: {self._speed:.1f}")
        print(f"Angle: {self._rotate_angle:.1f}")
        print(f"Height: {self._height:.1f}")
